src/joystick/haiku/SDL_haikujoystick.cc

```diff
@@ -113,7 +113,7 @@ extern "C"
 /* Function to perform the mapping from device index to the instance id for this index */
     static SDL_JoystickID HAIKU_JoystickGetDeviceInstanceID(int device_index)
     {
-        return device_index;
+        return device_index + 1;
```

Comment:
This should probably use SDL_GetNextObjectID() to allocate object IDs for joysticks. That's a more involved change though, what you're doing here will probably suffice for the short term.